pkg/component/extensions/operatingsystemconfig/init/bootstrap_config.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package init
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	nodeagentv1alpha1 "github.com/gardener/gardener/pkg/nodeagent/apis/config/v1alpha1"
+)
+
+func getBootstrapConfiguration(worker gardencorev1beta1.Worker) (*nodeagentv1alpha1.BootstrapConfiguration, error) {
+	bootstrapConfiguration := &nodeagentv1alpha1.BootstrapConfiguration{}
+
+	kubeletDataVolumeSize, err := getKubeletDataVolumeSize(worker)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting kubelet data volume size: %w", err)
+	}
+	bootstrapConfiguration.KubeletDataVolumeSize = kubeletDataVolumeSize
```

Comment:
Nit. We could make it a bit more concise.

```suggestion
	bootstrapConfiguration.KubeletDataVolumeSize, err := getKubeletDataVolumeSize(worker)
	if err != nil {
		return nil, fmt.Errorf("failed getting kubelet data volume size: %w", err)
	}
```